python/virgo/mpi/test_parallel_hdf5.py

```diff
@@ -449,6 +472,19 @@ def test_multi_file_one_file_per_rank_group(tmp_path):
         do_multi_file_test(tmp_path, basename="file_per_rank_group", nr_files=comm_size, elements_per_file=n,
                            group="group")
 
+@pytest.mark.mpi
+def test_inconsistent_dtype(tmp_path):
+
+    from mpi4py import MPI
+    comm = MPI.COMM_WORLD
+    comm_size = comm.Get_size()
+
+    for n in (2, 10, 100, 1000, 10000):
+        with pytest.raises(RuntimeError):
+            do_multi_file_test(tmp_path, basename="inconsistent_dtype", nr_files=comm_size,
```

Comment:
This raises an AssertionError if run on one rank. pytest calls that a failure because it's expecting a RuntimeError.

Reply:
On one rank we can still trigger a failure with multiple files. I'll add separate independent and collective tests and we can skip the collective version on one rank.